File: 657-robot-return-to-origin/main.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        
        int position[2] = {0, 0};

        for (char move : moves) {
            switch (move) {
                case 'U':
                    position[0]++;
                    break;
                case 'D':
                    position[0]--;
                    break;
                case 'L':
                    position[1]--;
                    break;
                case 'R':
                    position[1]++;
                    break;
            }
        }

        return position[0] == 0 && position[1] == 0;
    }
};

File: 657-robot-return-to-origin/main.go
func judgeCircle(moves string) bool {
    
    position := [2]int{0, 0}

    for _, move := range moves {
        if move == 'U' {
            position[0]++
        }else if move == 'D' {
            position[0]--
        }else if move == 'L'{
            position[1]--
        }else {
            position[1]++
        }
    }

    return position == [2]int{0, 0}
}
